src/ConfigCatProvider.tsx

```diff
@@ -103,95 +109,83 @@ function serverContextNotSupported(): Error {
   );
 }
 
+function providerIdText(providerId?: string) {
+  return providerId ? ` with id="${providerId}"` : " without id attribute";
+}
+
+export function createConfigCatProviderError(methodName: string, providerId?: string): Error {
+  return Error(`${methodName} must be used in ConfigCatProvider${providerIdText(providerId)}!`);
+}
+
 class ConfigCatClientStub implements IConfigCatClient {
-  readonly isOffline = true;
+  readonly isOffline: boolean;
 
-  getValueAsync<T extends SettingValue>(_key: string, _defaultValue: T, _user?: User | undefined): Promise<SettingTypeOf<T>> {
+  getValueAsync<T extends SettingValue>(_key: string, _defaultValue: T, _user?: User): Promise<SettingTypeOf<T>> {
     throw serverContextNotSupported();
   }
-
-  getValueDetailsAsync<T extends SettingValue>(_key: string, _defaultValue: T, _user?: User | undefined): Promise<IEvaluationDetails<SettingTypeOf<T>>> {
+  getValueDetailsAsync<T extends SettingValue>(_key: string, _defaultValue: T, _user?: User): Promise<IEvaluationDetails<SettingTypeOf<T>>> {
     throw serverContextNotSupported();
   }
-
   getAllKeysAsync(): Promise<string[]> {
     throw serverContextNotSupported();
   }
-
-  getAllValuesAsync(_user?: User | undefined): Promise<SettingKeyValue<SettingValue>[]> {
+  getAllValuesAsync(_user?: User): Promise<SettingKeyValue<SettingValue>[]> {
     throw serverContextNotSupported();
   }
-
-  getAllValueDetailsAsync(_user?: User | undefined): Promise<IEvaluationDetails<SettingValue>[]> {
+  getAllValueDetailsAsync(_user?: User): Promise<IEvaluationDetails<SettingValue>[]> {
     throw serverContextNotSupported();
   }
-
   getKeyAndValueAsync(_variationId: string): Promise<SettingKeyValue<SettingValue> | null> {
     throw serverContextNotSupported();
   }
-
   forceRefreshAsync(): Promise<RefreshResult> {
     throw serverContextNotSupported();
   }
-
   waitForReady(): Promise<ClientCacheState> {
     throw serverContextNotSupported();
   }
-
   snapshot(): IConfigCatClientSnapshot {
     throw serverContextNotSupported();
   }
-
   setDefaultUser(_defaultUser: User): void {
     throw serverContextNotSupported();
   }
-
   clearDefaultUser(): void {
     throw serverContextNotSupported();
   }
-
   setOnline(): void {
     throw serverContextNotSupported();
   }
-
   setOffline(): void {
     throw serverContextNotSupported();
   }
-
-  dispose(): void { }
-
+  dispose(): void {
+    throw serverContextNotSupported();
+  }
```

Comment:
Let's revert this change.

```suggestion
dispose(): void { }
```